chess-AI/experimental-work/train.py

```diff
@@ -0,0 +1,81 @@
+import random
+import chess
+from mcts import Mcts
+from nn_layout import PlayNetwork
+from output_representation import PlayNetworkPolicyConverter
+
+class MctsTrain: # algorithm for training the model.
+
+    def __init__(self,mcts_simulations):
+
+        # Set the starting fen_string
+        self.start_fen = ("rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP"
+        "/RNBQKBNR w KQkq - 0 1")
+
+        # Set the number of Monte Carlo Simulations
+        self.mcts_simulations = mcts_simulations
+        self.training_examples = []
+
+    def training_episode(self,nnet):
+
+        # At each episode need to set the training example to empty
+        self.training_examples = []
+
+        # The board needs to be initialized ot the starting state
+        board = chess.Board(self.start_fen)
+        fen_string = self.start_fen
+        mcts = Mcts() # assign MCTS object
+        policy_converter = PlayNetworkPolicyConverter()
+
+        while True:
+
+            # Perform mcts simulation
+            for _ in range(self.mcts_simulations):
+                mcts.search(board,nnet)
+
+            # Need to get the moves and policy from the mcts
+            keys,policy = mcts.find_search_probs(fen_string)
+
+            # Makes a random choice from the policy
+            choice = random.randint(0,len(policy) - 1)
+            action = keys[choice]
+
+            # Converts mcts search probabilites to (8,8,73) vector
+            policy = policy_converter.compute_full_search_probs(keys, policy, board)
```

Comment:
instead of overwriting the other variable here (which makes it hard to follow) you can create a new variable called `full_search_probs`.